lib/ui/common/ActionSheetListItem.dart

```diff
@@ -0,0 +1,175 @@
+          fontWeight: FontWeight.normal,
+          color: Color(0xFFff6060)),
+      actionItemEdgePadding: const EdgeInsets.symmetric(horizontal: 32),
+      actionItemHeight: 70,
+      iconLineSpace: 21.5,
+      actionItemElevation: 0,
+      iconHeight: 20,
+      maxLines: 1,
+    );
+  }
+
+  factory ActionSheetListItemStyle.selectableStyle() {
+    return ActionSheetListItemStyle.defaultStyle().copyWith(iconHeight: 24);
+  }
+
+  ActionSheetListItemStyle copyWith(
+      {Color actionItemBackgroundColor,
+      TextStyle actionTextStyle,
+      TextStyle destructiveTextStyle,
+      EdgeInsets actionItemEdgePadding,
+      double actionItemHeight,
+      double iconLineSpace,
+      double actionItemElevation,
+      double iconHeight,
+      int maxLines}) {
+    return ActionSheetListItemStyle(
+        actionItemBackgroundColor:
+            actionItemBackgroundColor ?? this.actionItemBackgroundColor,
+        actionTextStyle: actionTextStyle ?? this.actionTextStyle,
+        destructiveTextStyle: destructiveTextStyle ?? this.destructiveTextStyle,
+        actionItemEdgePadding:
+            actionItemEdgePadding ?? this.actionItemEdgePadding,
+        actionItemHeight: actionItemHeight ?? this.actionItemHeight,
+        iconLineSpace: iconLineSpace ?? this.iconLineSpace,
+        actionItemElevation: actionItemElevation ?? this.actionItemElevation,
+        iconHeight: iconHeight ?? this.iconHeight,
+        maxLines: maxLines ?? this.maxLines);
+  }
+}
+
+class ActionSheetListItem extends StatelessWidget {
+  final ActionSheetListItemViewModel _viewModel;
+  final ActionSheetListItemStyle _style;
+  final int _numberOfItems;
+  final int _currentItem;
+
+  const ActionSheetListItem(
+      {Key key,
+      ActionSheetListItemViewModel viewModel,
+      ActionSheetListItemStyle style,
+      int numberOfActions,
+      int currentAction})
+      : this._viewModel = viewModel,
+        this._style = style,
+        this._numberOfItems = numberOfActions,
+        this._currentItem = currentAction,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        Card(
+          elevation: _style.actionItemElevation,
+          color: _style.actionItemBackgroundColor,
+          child: InkWell(
+            onTap: () => _viewModel.action,
+            child: Container(
+              padding: _style.actionItemEdgePadding,
+              alignment: Alignment.center,
+              height: _style.actionItemHeight,
+              child: Wrap(
+                  direction: Axis.horizontal,
+                  crossAxisAlignment: WrapCrossAlignment.start,
+                  children: <Widget>[
+                    Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: _buildActionContent(_style, _viewModel)),
+                  ]),
+            ),
+          ),
+        ),
+        _currentItem == _numberOfItems - 1
```

Comment:
get the style to decide this. eg. "hasDivider" The list widget should then have this logic and you won't need to pass in the number of items etc.

Reply:
The divider shouldn't be inside the item so with the static list @jessicahernandezhv will add these divider as static

Reply:
Divider out of the item, done